backend/src/main/java/com/crm/service/impl/CarServiceImpl.java

```diff
@@ -49,4 +54,32 @@ public CarResponse getCarByRegistrationNumber(final String registrationNumber) {
                 .map(carMapper::convertToDto)
                 .orElseThrow(() -> new CarNotFoundException(ErrorDict.REGISTRATION_NUMBER_NOT_FOUND));
     }
+
+    @Override
+    public CarEntity addCar(CarRequest carRequest) {
+
+        CarTypeEntity carTypeEntity;
+        String nameOfCarTypeEntity = carRequest.getCarTypeEntity().getName();
+
+        if (carRepository.findByRegistrationNumberIgnoreCase(carRequest.getRegistrationNumber()).isPresent()) {
+            throw new CarHandlingException(ErrorDict.CAR_CREATE_REGISTRATION_NUMBER_EXISTS);
+        }
+
+        if (carRepository.findByVinIgnoreCase(carRequest.getVin()).isPresent()) {
+            throw new CarHandlingException(ErrorDict.CAR_CREATE_VIN_EXISTS);
+        }
+
+        if (nameOfCarTypeEntity.isBlank()) {
+            carTypeEntity = carTypeRepository.findByNameIgnoreCase("Other")
+                    .orElseGet(() -> CarTypeEntity.builder().name("Other").build());
+        } else {
+            carTypeEntity = carTypeRepository.findByNameIgnoreCase(nameOfCarTypeEntity)
+                    .orElseGet(() -> CarTypeEntity.builder().name(nameOfCarTypeEntity).build());
+        }
+
+        CarEntity newCar = carMapper.convertToEntity(carRequest);
+        newCar.setCarTypeEntity(carTypeEntity);
+
+        return carRepository.save(newCar);
```

Comment:
powinniśmy zwrócić CarResponse, czyli jeszcze by wypadało zmapować ten obiekt na Dto (zobacz u mnie jak to zaimplementowałem)